Status segment never detecting errors in pwsh 7.5.4?

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Hi Jan, many thanks for your superb theming tool which I've used for years.  I was using the PowerShell module and hadn't updated for ages, have just moved up to latest 29.4.1 and updated my custom JSON, but I'm hitting an issue with my `status` segment.

Whatever I try, it seems the `.Code` and `.Error` properties are not picking up on failures.  This is true whether it's a pwsh cmdlet or an executable that fails.  I've copied the exact block from a couple of your example profiles and they behave the same (blueish and iterm2).

So with this segment for example:

```
        {
          "background": "#193549",
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "options": {
            "always_enabled": true
          },
          "style": "powerline",
          "template": " {{ .Error }} {{ .Code }} ",
          "type": "status"
        }
```

With my custom theme I see this:

<img width="1313" height="392" alt="Image" src="https://github.com/user-attachments/assets/441f68eb-5185-45a6-a361-d8a2fb697b54" />

Or, just for 100% confirmation, with blueish totally unmodified:

<img width="2091" height="367" alt="Image" src="https://github.com/user-attachments/assets/1f15e9ae-e364-42b2-856e-dbcf3f44c1cd" />

### Theme

Blueish or my own custom theme

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
(base)  Mike@AgentSmith  18:31:07  ~\code  pwsh  false 
❯ oh-my-posh debug --plain

Version: 29.4.1

Shell: shell (7.5.4)

Prompt:

 Mike@AgentSmith  18:32:43  ~\code  shell  false 
❯

Segments:

ConsoleTitle(false)                        -   0 ms
Session(true)                              -   0 ms
Time(true)                                 -   0 ms
Executiontime(false)                       -   0 ms
Path(true)                                 -   1 ms
Dotnet(false)                              -   1 ms
Git(false)                                 -   1 ms
Shell(true)                                -   0 ms
Root(false)                                -   0 ms
Status(true)                               -   1 ms
Text(true)                                 -   0 ms

Run duration: 3.6634ms

Cache path: C:\Users\Mike\AppData\Local\Packages\ohmyposh.cli_96v55e8n804z4\LocalCache\Local\oh-my-posh

Config path: C:\Users\Mike\OneDrive\Tech\working-env-config\PowerShell\oh-my-posh\Themes\.go-my-posh.json

Logs:

[DEBUG] 18:32:43.680 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 18:32:43.680 store.go:]:141 → (session) key not found: prompt_count_cache
[TRACE] 18:32:43.680 store.go:](session prompt_count_cache) - 0s
[TRACE] 18:32:43.680 terminal.go:setPromptCount() - 0s
[TRACE] 18:32:43.680 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.680 terminal.go:setPwd:102 → C:\Users\Mike\code
[TRACE] 18:32:43.680 terminal.go:setPwd() - 0s
[TRACE] 18:32:43.680 terminal.go:Init() - 0s
[DEBUG] 18:32:43.681 path_windows.go:PackageFamilyName:49 → PackageFamilyName: ohmyposh.cli_96v55e8n804z4
[TRACE] 18:32:43.681 path_windows.go:PackageFamilyName() - 524.4µs
[TRACE] 18:32:43.681 path.go:Path() - 524.4µs
[DEBUG] 18:32:43.681 store.go:init:94 → (device) loading DSC_SHELL
[DEBUG] 18:32:43.681 store.go:init:94 → (device) loading INITVERSIONPWSH
[DEBUG] 18:32:43.681 store.go:init:94 → (device) loading DSC_CONFIGURATION
[TRACE] 18:32:43.681 store.go:init(device omp.cache) - 524.4µs
[TRACE] 18:32:43.681 init.go:SessionID() - 0s
[TRACE] 18:32:43.681 path.go:Path() - 0s
[DEBUG] 18:32:43.682 store.go:init:94 → (session) loading prompt_count_cache
[DEBUG] 18:32:43.682 store.go:init:94 → (session) loading CONFIG
[DEBUG] 18:32:43.682 store.go:init:94 → (session) loading toggle_cache
[DEBUG] 18:32:43.682 store.go:init:94 → (session) loading CONFIG_SOURCE
[TRACE] 18:32:43.682 store.go:init(session pwsh.3274701f-ccb5-418c-b389-3354d27a4ad4.omp.cache) - 520µs
[DEBUG] 18:32:43.682 store.go:]:141 → (device) key not found: reload
[TRACE] 18:32:43.682 store.go:](device reload) - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: CONFIG - /gIHfwMBAQZDb25maWcB/4AAASABB1BhbGV0dGUB/4IAAQtEZWJ1Z1Byb21wdAH/hAABA1ZhcgH/jgABCFBhbGV0dGVzAf+QAAEJVmFsaWRMaW5lAf+EAAEPU2Vjb25kYXJ5UHJvbXB0Af+EAAEPVHJhbnNpZW50UHJvbXB0Af+EAAEJRXJyb3JMaW5lAf+EAAEETWFwcwH/lAABB1VwZ3JhZGUB/5gAAQdFeHRlbmRzAQwAAQtBY2NlbnRDb2xvcgEMAAEUQ29uc29sZVRpdGxlVGVtcGxhdGUBDAABA1BXRAEMAAEGU291cmNlAQwAAQZGb3JtYXQBDAABElRlcm1pbmFsQmFja2dyb3VuZAEMAAEOVG9vbFRpcHNBY3Rpb24BDAABBkJsb2NrcwH/ngABBUN5Y2xlAf+iAAENSVRlcm1GZWF0dXJlcwH/pAABCFRvb2x0aXBzAf+cAAEHVmVyc2lvbgEEAAENTWlncmF0ZUdseXBocwECAAEFQXN5bmMBAgABEFNoZWxsSW50ZWdyYXRpb24BAgABCkZpbmFsU3BhY2UBAgABDVVwZ3JhZGVOb3RpY2UBAgABDlBhdGNoUHdzaEJsZWVkAQIAAQtBdXRvVXBncmFkZQECAAEXRW5hYmxlQ3Vyc29yUG9zaXRpb25pbmcBAgABCVN0cmVhbWluZwEEAAAAF/+BBAEBB1BhbGV0dGUB/4IAAQwBDAAA/gIx/4MDAQEHU2VnbWVudAH/hAABJAEHT3B0aW9ucwH/hgABClByb3BlcnRpZXMB/4YAAQVDYWNoZQH/iAABBUFsaWFzAQwAAQ5MZWFkaW5nRGlhbW9uZAEMAAEPVHJhaWxpbmdEaWFtb25kAQwAAQhUZW1wbGF0ZQEMAAEKRm9yZWdyb3VuZAEMAAEOVGVtcGxhdGVzTG9naWMBDAABD1Bvd2VybGluZVN5bWJvbAEMAAEKQmFja2dyb3VuZAEMAAEGRmlsbGVyAQwAAQRUeXBlAQwAAQVTdHlsZQEMAAEWTGVhZGluZ1Bvd2VybGluZVN5bWJvbAEMAAELUGxhY2Vob2xkZXIBDAABBFRpcHMB/4oAARNCYWNrZ3JvdW5kVGVtcGxhdGVzAf+MAAEJVGVtcGxhdGVzAf+MAAEORXhjbHVkZUZvbGRlcnMB/4oAAQ5JbmNsdWRlRm9sZGVycwH/igABBU5lZWRzAf+KAAETRm9yZWdyb3VuZFRlbXBsYXRlcwH/jAABBUluZGV4AQQAAQhNaW5XaWR0aAEEAAEIRHVyYXRpb24BBAABCk5hbWVMZW5ndGgBBAABCE1heFdpZHRoAQQAAQdUaW1lb3V0AQQAAQdOZXdsaW5lAQIAAQdFbmFibGVkAQIAAQ9JbnZlcnRQb3dlcmxpbmUBAgABBUZvcmNlAQIAAQdUb2dnbGVkAQIAAQdQZW5kaW5nAQIAAQtJbnRlcmFjdGl2ZQECAAAAE/+FBAEBA01hcAH/hgABDAEQAAAt/4cDAQEFQ2FjaGUB/4gAAQIBCER1cmF0aW9uAQwAAQhTdHJhdGVneQEMAAAAFv+JAgEBCFtdc3RyaW5nAf+KAAEMAAAS/4sCAQEETGlzdAH/jAABDAAAJ/+NBAEBF21hcFtzdHJpbmddaW50ZXJmYWNlIHt9Af+OAAEMARAAAC3/jwMBAQhQYWxldHRlcwH/kAABAgEETGlzdAH/kgABCFRlbXBsYXRlAQwAAAAp/5EEAQEYbWFwW3N0cmluZ11jb2xvci5QYWxldHRlAf+SAAEMAf+CAAA//5MDAQEGQ29uZmlnAf+UAAEDAQhVc2VyTmFtZQH/lgABCEhvc3ROYW1lAf+WAAEJU2hlbGxOYW1lAf+WAAAAE/+VBAEBA01hcAH/lgABDAEMAABc/5cDAQEGQ29uZmlnAf+YAAEGAQZTb3VyY2UBDAABCEludGVydmFsAQwAAQZMYXRlc3QBDAABBEF1dG8BAgABDURpc3BsYXlOb3RpY2UBAgABBUZvcmNlAQIAAAAe/50CAQEPW10qY29uZmlnLkJsb2NrAf+eAAH/mgAA/5D/mQMBAv+aAAEKAQRUeXBlAQwAAQlBbGlnbm1lbnQBDAABBkZpbGxlcgEMAAEIT3ZlcmZsb3cBDAABDkxlYWRpbmdEaWFtb25kAQwAAQ9UcmFpbGluZ0RpYW1vbmQBDAABCFNlZ21lbnRzAf+cAAEHTmV3bGluZQECAAEFRm9yY2UBAgABBUluZGV4AQQAAAAg/5sCAQERW10qY29uZmlnLlNlZ21lbnQB/5wAAf+EAAAU/6ECAQEFQ3ljbGUB/6IAAf+gAAAq/58DAQL/oAABAgEKQmFja2dyb3VuZAEMAAEKRm9yZWdyb3VuZAEMAAAAG/+jAgEBDUlUZXJtRmVhdHVyZXMB/6QAAQwAAP4C2/+ACgEDY2RuAQQxNjhoAAVcQzpcVXNlcnNcTWlrZVxPbmVEcml2ZVxUZWNoXHdvcmtpbmctZW52LWNvbmZpZ1xQb3dlclNoZWxsXG9oLW15LXBvc2hcVGhlbWVzXC5nby1teS1wb3NoLmpzb24BBGpzb24DAQEGcHJvbXB0AQRsZWZ0BQoIByNmZmZmZmYDByNjMzg2ZjECB3Nlc3Npb24BCXBvd2VybGluZQED7oK2AAEBC3RpbWVfZm9ybWF0BnN0cmluZwwKAAgxNTowNDowNQcHI2ZmZmZmZgID7oKwAQcjQTM0OUE0AgR0aW1lAQlwb3dlcmxpbmUAAQIFc3R5bGUGc3RyaW5nDAgABmF1c3Rpbgl0aHJlc2hvbGQHZmxvYXQ2NAgFAP1Af0AHByNmZmZmZmYCA+6CsAEHIzg4MDBkZAINZXhlY3V0aW9udGltZQEJcG93ZXJsaW5lAAEBBXN0eWxlBnN0cmluZwwGAARmdWxsBwcjZmZmZmZmAgPugrABByM0OTM1QzgCBHBhdGgBCXBvd2VybGluZQAIByNmZmZmZmYCA+6CsAEHIzU1NTVlZQIGZG90bmV0AQlwb3dlcmxpbmUAAQETZmV0Y2hfdXBzdHJlYW1faWNvbgRib29sAgIAAQcHI2ZmZmZmZgID7oKwAQcjNEI3MUU1AgNnaXQBCXBvd2VybGluZQAIByNmZmZmZmYCA+6CsAEHIzREQTFFMQIFc2hlbGwBCXBvd2VybGluZQAIByMxOTM1NDkCA+6CsAEHI2ZmZmY2NgIEcm9vdAEJcG93ZXJsaW5lAAEBDmFsd2F5c19lbmFibGVkBGJvb2wCAgABBg4ge3sgLkVycm9yIH19IAEHI2ZmZmZmZgID7oKwAQcjMTkzNTQ5AgZzdGF0dXMBCXBvd2VybGluZQAHBQrina8gAQcjYWFhYWFhBQR0ZXh0AQVwbGFpbgAABAQA
[TRACE] 18:32:43.682 store.go:](session CONFIG) - 0s
[TRACE] 18:32:43.682 gob.go:Restore() - 0s
[TRACE] 18:32:43.682 gob.go:Get() - 0s
[TRACE] 18:32:43.682 terminal.go:Shell() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 store.go:]:141 → (session) key not found: template_cache
[TRACE] 18:32:43.682 store.go:](session template_cache) - 0s
[TRACE] 18:32:43.682 cache.go:restoreCache() - 0s
[TRACE] 18:32:43.682 terminal_windows.go:Root() - 0s
[TRACE] 18:32:43.682 terminal.go:Shell() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:StatusCodes() - 0s
[TRACE] 18:32:43.682 terminal_windows.go:IsWsl() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal_windows.go:IsWsl() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 terminal.go:User:236 → Mike
[TRACE] 18:32:43.682 terminal.go:User() - 0s
[DEBUG] 18:32:43.682 terminal.go:Host:253 → AgentSmith
[TRACE] 18:32:43.682 terminal.go:Host() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 terminal.go:Getenv:64 → NO DATA
[TRACE] 18:32:43.682 terminal.go:Getenv(SHLVL) - 0s
[DEBUG] 18:32:43.682 writer.go:Init:107 → terminal program: Windows Terminal
[DEBUG] 18:32:43.682 writer.go:Init:108 → terminal shell: shell
[DEBUG] 18:32:43.682 terminal.go:Getenv:64 → NO DATA
[TRACE] 18:32:43.682 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[DEBUG] 18:32:43.682 store.go:]:141 → (device) key not found: accent_color
[TRACE] 18:32:43.682 store.go:](device accent_color) - 0s
[DEBUG] 18:32:43.682 terminal_windows.go:WindowsRegistryKeyValue:205 → ColorizationColor(DWORD): 0x003A3A3A
[TRACE] 18:32:43.682 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 18:32:43.682 colors_windows.go:GetAccentColor() - 0s
[TRACE] 18:32:43.682 colors.go:SetAccentColor() - 0s
[TRACE] 18:32:43.682 terminal.go:Shell() - 0s
[DEBUG] 18:32:43.682 terminal.go:Getenv:64 → 7.5.4
[TRACE] 18:32:43.682 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 18:32:43.682 debug.go:PrintDebug:24 → segment: Title
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 terminal.go:Getenv:64 → 12
[TRACE] 18:32:43.682 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 18:32:43.682 terminal.go:Getenv:64 → 8
[TRACE] 18:32:43.682 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Text
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Text): 14) - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Session): 5) - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Path
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Path): 8) - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Time): 6) - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 map.go:Bool:133 → display_cygpath: false
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Executiontime
[TRACE] 18:32:43.682 separator.go:Separator() - 0s
[TRACE] 18:32:43.682 terminal.go:Shell() - 0s
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 map.go:String:83 → time_format: 15:04:05
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Dotnet
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Executiontime): 7) - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[DEBUG] 18:32:43.682 text.go:patchTemplate:149 →  {{ if .Data.SSHSession }} {{ end }}{{ .UserName }}@{{ .HostName }}
[DEBUG] 18:32:43.682 map.go:Bool:133 → always_enabled: false
[TRACE] 18:32:43.682 terminal.go:ExecutionTime() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Dotnet): 9) - 0s
[DEBUG] 18:32:43.682 map.go:Float64:155 → threshold: 500.000000
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Root
[DEBUG] 18:32:43.682 map.go:Bool:133 → mapped_locations_enabled: true
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[DEBUG] 18:32:43.682 map.go:String:79 → windows_registry_icon: 
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 map.go:String:79 → windows_registry_icon: 
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 separator.go:Separator() - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Root): 12) - 0s
[DEBUG] 18:32:43.682 map.go:StringArray:220 → extensions: [*.cs *.csx *.vb *.sln *.slnx *.slnf *.csproj *.vbproj *.fs *.fsx *.fsproj global.json]
[DEBUG] 18:32:43.682 map.go:StringArray:220 → folders: []
[DEBUG] 18:32:43.682 map.go:Bool:133 → home_enabled: false
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[DEBUG] 18:32:43.682 map.go:String:79 → display_mode: files
[TRACE] 18:32:43.682 terminal_windows.go:Root() - 0s
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Shell
[DEBUG] 18:32:43.682 home.go:12 → C:\Users\Mike
[DEBUG] 18:32:43.682 map.go:String:79 → home_icon: ~
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 segment.go:Execute:173 → segment: Status
[DEBUG] 18:32:43.682 store.go:]:154 → (session) found entry: toggle_cache - map[]
[DEBUG] 18:32:43.682 map.go:KeyValueMap:208 → mapped_locations: map[]
[TRACE] 18:32:43.682 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[DEBUG] 18:32:43.682 segment.go:isToggled:291 → no toggles found
[TRACE] 18:32:43.682 separator.go:Separator() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 separator.go:Separator() - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Status): 13) - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 terminal.go:StatusCodes() - 0s
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.682 map.go:String:79 → status_template: {{ .Code }}
[TRACE] 18:32:43.682 terminal.go:GOOS() - 0s
[TRACE] 18:32:43.682 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Shell): 11) - 0s
[DEBUG] 18:32:43.682 map.go:KeyValueMap:208 → mapped_shell_names: map[]
[TRACE] 18:32:43.682 terminal.go:Shell() - 0s
[DEBUG] 18:32:43.682 map.go:String:79 → gitdir_format:
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[TRACE] 18:32:43.682 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.682 map.go:StringArray:220 → cycle: []
[DEBUG] 18:32:43.682 map.go:String:79 → folder_separator_template:
[DEBUG] 18:32:43.682 map.go:String:79 → folder_separator_icon: \
[DEBUG] 18:32:43.682 map.go:Bool:133 → cycle_folder_separator: false
[DEBUG] 18:32:43.682 map.go:String:79 → folder_format: %s
[TRACE] 18:32:43.683 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.683 text.go:patchTemplate:149 → {{ .Code }}
[DEBUG] 18:32:43.683 map.go:String:79 → edge_format: %s
[DEBUG] 18:32:43.683 map.go:String:79 → left_format: %s
[DEBUG] 18:32:43.683 map.go:String:79 → right_format: %s
[DEBUG] 18:32:43.683 home.go:12 → C:\Users\Mike
[TRACE] 18:32:43.683 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.683 segment.go:Execute:173 → segment: Git
[DEBUG] 18:32:43.683 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 18:32:43.683 store.go:](session toggle_cache) - 0s
[DEBUG] 18:32:43.683 segment.go:isToggled:291 → no toggles found
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:Flags() - 0s
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.csx) - 0s
[TRACE] 18:32:43.683 terminal.go:GOOS() - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:StackCount() - 0s
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.vb) - 0s
[DEBUG] 18:32:43.683 map.go:Bool:133 → disable_with_jj: false
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.sln) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.slnx) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.slnf) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.csproj) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.vbproj) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.fs) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.fsx) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(*.fsproj) - 0s
[DEBUG] 18:32:43.683 terminal.go:HasFilesInDir:151 → false
[TRACE] 18:32:43.683 terminal.go:HasFilesInDir(global.json) - 0s
[DEBUG] 18:32:43.683 map.go:StringArray:220 → tooling: [dotnet]
[TRACE] 18:32:43.683 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.683 win32_windows.go:isWriteable:259 → not current user or in group
[DEBUG] 18:32:43.683 terminal.go:ResolveSymlink:195 → C:\Users\Mike\code
[DEBUG] 18:32:43.683 win32_windows.go:isWriteable:263 → current user is member of S-1-5-32-544
[TRACE] 18:32:43.683 terminal.go:ResolveSymlink(C:\Users\Mike\code) - 527.7µs
[DEBUG] 18:32:43.683 win32_windows.go:isWriteable:271 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 18:32:43.683 win32_windows.go:isWriteable:273 → user has write access
[TRACE] 18:32:43.683 terminal_windows.go:DirIsWritable() - 527.7µs
[TRACE] 18:32:43.683 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.683 text.go:patchTemplate:149 →  {{ .Data.CurrentDate | date .Data.Format }}
[DEBUG] 18:32:43.684 map.go:Bool:141 → always_enabled: true
[TRACE] 18:32:43.684 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.684 text.go:patchTemplate:149 →  {{ .Data.Path }}
[ERROR] 18:32:43.684 terminal.go:HasParentFilePath:464 → GetFileAttributesEx .git: The system cannot find the file specified.
[TRACE] 18:32:43.684 terminal.go:HasParentFilePath(.git) - 1.0501ms
[TRACE] 18:32:43.684 terminal.go:Flags() - 0s
[DEBUG] 18:32:43.684 text.go:patchTemplate:149 →  {{ .Data.Name }}
[DEBUG] 18:32:43.684 text.go:patchTemplate:149 →  {{ .Data.Error }}
[TRACE] 18:32:43.684 terminal.go:Shell() - 0s
[TRACE] 18:32:43.684 terminal.go:Shell() - 0s
[TRACE] 18:32:43.684 path.go:Path() - 0s
```

--- a/src/runtime/terminal_unix.go
+++ b/src/runtime/terminal_unix.go
@@ -110,15 +110,25 @@ func (term *Terminal) Platform() string {
 	}
 
 	platform, _, _, _ = host.PlatformInformation()
-	if platform == "arch" {
-		// validate for Manjaro
-		lsbInfo := term.FileContent("/etc/lsb-release")
-		if strings.Contains(strings.ToLower(lsbInfo), "manjaro") {
-			platform = "manjaro"
-		}
-	}
+	platform = term.getSpecialLinuxDistros(platform)
 
 	log.Debug(platform)
+	return platform
+}
+
+func (term *Terminal) getSpecialLinuxDistros(platform string) string {
+	lsbInfo := term.FileContent("/etc/lsb-release")
+
+	if platform == "arch" && strings.Contains(strings.ToLower(lsbInfo), "manjaro") {
+		// validate for Manjaro
+		return "manjaro"
+	}
+
+	if platform == "debian" && strings.Contains(strings.ToLower(lsbInfo), "zorin") {
+		// validate for Zorin OS
+		return "zorin"
+	}
+
 	return platform
 }
 

--- a/src/segments/os.go
+++ b/src/segments/os.go
@@ -68,12 +68,12 @@ func (oi *Os) getDistroIcon(distro string) string {
 		"elementary":          "\uf309",
 		"endeavouros":         "\uf322",
 		"fedora":              "\uf30a",
-		"freebsd":             "\u000f08e0",
+		"freebsd":             "\U000f08e0",
 		"gentoo":              "\uf30d",
 		"kali":                "\uf327",
 		"mageia":              "\uf310",
 		"manjaro":             "\uf312",
-		"mint":                "\u000f08ed",
+		"mint":                "\U000f08ed",
 		"neon":                "\uf331",
 		"nixos":               "\uf313",
 		"opensuse":            "\uf314",
@@ -85,6 +85,7 @@ func (oi *Os) getDistroIcon(distro string) string {
 		"slackware":           "\uf319",
 		"ubuntu":              "\uf31b",
 		"void":                "\uf32e",
+		"zorin":               "\uf32f",
 	}
 
 	if icon, ok := iconMap[distro]; ok {

--- a/src/shell/scripts/omp.ps1
+++ b/src/shell/scripts/omp.ps1
@@ -1,4 +1,4 @@
-# remove any existing dynamic module of OMP
+﻿# remove any existing dynamic module of OMP
 if ($null -ne (Get-Module -Name "oh-my-posh-core")) {
     Remove-Module -Name "oh-my-posh-core" -Force
 }
@@ -382,18 +382,6 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
     }
 
     $promptFunction = {
-        # Only return cached prompt if we're in a streaming redraw (RUNNING state)
-        # AND it's not a transient prompt. Don't use cached prompt for FINAL state
-        # as that means the previous prompt is complete and we need a fresh one.
-        if ($script:PromptType -ne 'transient' -and $script:Streaming.State -ne 'NEW') {
-            # Update ExtraPromptLineCount for PSReadLine to properly clear previous prompt
-            Set-PSReadLineOption -ExtraPromptLineCount (($script:Streaming.Prompt | Measure-Object -Line).Lines - 1)
-            return $script:Streaming.Prompt
-        }
-
-        # Stop any previous streaming process and reset state
-        Stop-StreamingProcess
-
         # store the original last command execution status
         if ($global:NVS_ORIGINAL_LASTEXECUTIONSTATUS -is [bool]) {
             # make it compatible with NVS auto-switching, if enabled
@@ -405,6 +393,18 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
 
         # store the original last exit code
         $script:OriginalLastExitCode = $global:LASTEXITCODE
+
+        # Only return cached prompt if we're in a streaming redraw (RUNNING state)
+        # AND it's not a transient prompt. Don't use cached prompt for FINAL state
+        # as that means the previous prompt is complete and we need a fresh one.
+        if ($script:PromptType -ne 'transient' -and $script:Streaming.State -ne 'NEW') {
+            # Update ExtraPromptLineCount for PSReadLine to properly clear previous prompt
+            Set-PSReadLineOption -ExtraPromptLineCount (($script:Streaming.Prompt | Measure-Object -Line).Lines - 1)
+            return $script:Streaming.Prompt
+        }
+
+        # Stop any previous streaming process and reset state
+        Stop-StreamingProcess
 
         # Reset tooltip command.
         $script:TooltipCommand = ''

--- a/website/docs/segments/system/os.mdx
+++ b/website/docs/segments/system/os.mdx
@@ -65,6 +65,7 @@ import Config from "@site/src/components/Config.js";
 | `slackware`           | `string`  |    `\uF319`    | the icon to use for Slackware Linux                      |
 | `ubuntu`              | `string`  |    `\uF31b`    | the icon to use for Ubuntu                               |
 | `void`                | `string`  |    `\uf32e`    | the icon to use for Void Linux                           |
+| `zorin`               | `string`  |    `\uf32f`    | the icon to use for Zorin OS                             |
 
 ## Template ([info][templates])
 

